src/main/java/net/dries007/tfc/common/recipes/ingredients/FluidItemIngredient.java

```diff
@@ -40,6 +42,35 @@ public boolean test(@Nullable ItemStack stack)
         return false;
     }
 
+    @Override
+    public ItemStack[] getItems()
+    {
+        dissolve();
+        return itemStacks;
+    }
+
+    @Override
+    protected void invalidate()
+    {
+        itemStacks = null;
+        super.invalidate();
+    }
+
+    private void dissolve()
+    {
+        if (this.itemStacks == null)
+        {
+            // how would you even handle the delegate here?
+            itemStacks = fluid.ingredient().getMatchingFluids().stream().map(fluid -> fluid.getBucket().getDefaultInstance()).toArray(ItemStack[]::new);
+        }
```

Comment:
This is mostly correct, however it should have two cases:
- if delegate is null, then return what you have here (list of buckets)
- if delegate is not null, call `delegate.getItems()`, and then filter by calling `test()` on each item stack.